content/blog/2020/05/2020-05-07-docker-windows-agents.adoc

```diff
@@ -0,0 +1,116 @@
+
+The agent image has a user `jenkins` which is used to run tasks on the Windows agent. The `jenkins` user is a member of the Administrators group inside the container.
+
+When the agent is created, it will show you the information needed to pass to agent.jar to run the agent and have it connect to the master. Here is an example of that command:
+
+```
+java -jar agent.jar -jnlpUrl https://JENKINS_URL/jenkins/computer/AGENTNAME/slave-agent.jnlp -secret SOMESECRETVALUE -workDir "C:\Users\jenkins\Agent"
+```
+
+This can be passed to the `docker run` command to connect the agent to the master (make sure to specify the correct values for JENKINS_URL, SOMESECRETVALUE, AGENTNAME and the workDir, which should match the value you specified in agent setup above).
+
+```
+docker run --name agent jenkins/agent:jdk8-nanoserver-1809 java -jar agent.jar -jnlpUrl https://JENKINS_URL/jenkins/computer/AGENTNAME/slave-agent.jnlp -secret SOMESECRETVALUE -workDir "C:\Users\jenkins\Agent"
+```
+
+You can also specify a volume for the work directory if desired (e.g., to keep certain files around in cached form or similar).
+
+```
+docker run -i --rm --name agent -v agent-workdir:C:/Users/jenkins/Work jenkins/agent java -jar C:/ProgramData/Jenkins/agent.jar -jnlpUrl https://JENKINS_URL/jenkins/computer/AGENTNAME/slave-agent.jnlp -secret SOMESECRETVALUE -workDir C:/Users/jenkins/Work
+```
+
+Just make sure this directory doesn't already exist or is empty inside the container (this is a requirement of volumes with Docker on Windows).
+
+
+
+=== jenkins/inbound-agent
+
+The inbound-agent Docker image tries to provide a higher level interaction with the agent.jar executable. It provides a PowerShell wrapper script around agent.jar and it is specified as the entrypoint so that you just need to pass in some command line arguments to run the agent. Several of these can also be passed using environment variables instead (they are listed in all capitals).
+
+Command line arguments include:
+
+* `-Url <URL>` or JENKINS_URL - the Jenkins master URL (this should NOT include the computer/NAME/slave-agent.jnlp part)
+* `-Secret <SECRET>` or JENKINS_SECRET - the secret as shown on the master after creating the agent
+* `-Name <NAME>` or JENKINS_AGENT_NAME - the name of the agent, it should match the name you specified when creating the agent on the master
+* `-Tunnel <HOST:PORT>` or JENKINS_TUNNEL - A tunnel host to route the TCP traffic through when connecting to the master
+* `-WorkDir <WORKDIR>` or JENKINS_AGENT_WORKDIR - same as the -workDir parameter mentioned above in the jenkins/agent image information.
+* `-WebSocket` or JENKINS_WEB_SOCKET=true - when present the connection to the master will be done via WebSocket instead of TCP networking.
```

Comment:
Since WebSocket is a layer over TCP, maybe better to describe the difference in terms of the term that is being used rather than the protocol?
```suggestion
* `-WebSocket` or JENKINS_WEB_SOCKET=true - when present the connection to the master will be done via WebSocket through the Jenkins URL rather than using a separate network port.
```